common/logisticspipes/gui/guidebook/GuiGuideBook.java

```diff
@@ -0,0 +1,779 @@
+package logisticspipes.gui.guidebook;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.client.renderer.BufferBuilder;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.MathHelper;
+
+import logisticspipes.items.ItemGuideBook;
+import logisticspipes.utils.string.StringUtils;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.lwjgl.opengl.GL11;
+
+import logisticspipes.LPConstants;
+import logisticspipes.gui.guidebook.book.MenuItem;
+import logisticspipes.utils.GuideBookContents;
+
+public class GuiGuideBook extends GuiScreen {
+
+	/*
+	 * TODO - Back button or bookmarks, whichever is more wanted
```

Comment:
Please no TODOs in code

Reply:
Yeah I've already set up a project so I'll remove that, it's the Glo I put in the first comment

Reply:
Sadly I seem to need to login to see the Glo. If you want us to see it, you could post/update a screenshot of that